app/controllers/admin/groups_controller.rb

```diff
@@ -24,13 +24,22 @@ def bulk
 
   def bulk_perform
     group = Group.find(params[:group_id].to_i)
+    users_added = 0
     if group.present?
       users = (params[:users] || []).map {|u| u.downcase}
-      user_ids = User.where("username_lower in (:users) OR email IN (:users)", users: users).pluck(:id)
-      group.bulk_add(user_ids) if user_ids.present?
+      valid_emails = valid_usernames = {}
+      valid_users = User.where("username_lower IN (:users) OR email IN (:users)", users: users).pluck(:id, :username_lower, :email)
+      valid_users.each do |vu|
```

Comment:
Awesome, thanks for the tip! Done via https://github.com/discourse/discourse/commit/e3f82140d8fa7f7e01378ccd5dcad57a96f2909b.